lib/rspec/rails/matchers/have_reported_error.rb

```diff
@@ -28,17 +33,20 @@ def report(error, **attrs)
       class HaveReportedError < RSpec::Rails::Matchers::BaseMatcher
         # Initialize the matcher following raise_error patterns
         #
+        # Uses UndefinedValue as default to distinguish between no argument 
+        # passed vs explicitly passed nil (same as raise_error matcher).
+        #
         # @param expected_error_or_message [Class, String, Regexp, nil] 
         #   Error class, message string, or message pattern
         # @param expected_message [String, Regexp, nil] 
         #   Expected message when first param is a class
-        def initialize(expected_error_or_message = nil, expected_message = nil)
+        def initialize(expected_error_or_message = UndefinedValue, expected_message = nil)
           @actual_error = nil
           @attributes = {}
           @error_subscriber = nil
           
           case expected_error_or_message
-          when nil
+          when nil, UndefinedValue
```

Comment:
We usually make a distinction between those two, where nil is typically a sign of a mistake. For some matchers, we warn if nil is passed. Do you think there can be legitimate cases of passing nil as an argument?

Reply:
Ok! Adjusted.